app/services/fixity_checker.rb

```diff
@@ -0,0 +1,117 @@
+# FixityChecker.new(asset).check
+# FixityChecker.new(asset).prune_checks
+# This first version of the fixity checker only checks
+# for actual discrepancies between already-recorded
+# checksums and calculated ones.
+#
+# This means:
+# Assets with nil files are ignored.
+# Files with nil checksums are ignored.
+
+class FixityChecker
+  attr_reader :asset
+
+  NUMBER_OF_RECENT_PASSED_CHECKS_TO_KEEP = 5
+  CHECK_CYCLE_LENGTH = 7
+
+
+  # @param work [Work] Work object, it's members will be put into a zip
+  # @param callback [proc], proc taking keyword arguments progress_i: and progress_total:, can
+  #   be used to update a progress UI.
+  def initialize(asset)
+    raise ArgumentError.new("Please pass in an Asset.") unless asset.is_a? Asset
+    @asset = asset
+  end
+
+  #FixityChecker.new(asset).check
+  def check(sieve_integer = 0)
+    return unless sift(sieve_integer)
+    return nil if @asset.nil?
+    return nil if @asset.file.nil?
+    the_expected_checksum = expected_checksum
+    # Note: @asset.file.sha512 can be nil on e.g.
+    # items imported with DISABLE_BYTESTREAM_IMPORT=true
+    # For now, we are ignoring these.
+    return nil if the_expected_checksum.nil?
```

Comment:
I think you should raise if you can't fixity check cause don't have a fixity on file, instead of silently ignoring/failing.